File: functions/java/src/main/java/functions/Post.kt
package functions

import java.util.*

data class Post(
    val userName: String = "",
    val userId: String = "",
    val userPhotoUrl: String = "",
    val text: String = "",
    val commentCount: Int = 0,
    val likedUserIds: List<String> = listOf(),
    val tags: List<String> = listOf(),
    val createdDate: Date = Date(),
    val timeSlots: List<Date> = listOf(),
    val images: List<String> = listOf(),
    val id: String = "",
)

File: domain/lib/post/post.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'post.freezed.dart';

@freezed
class Post with _$Post {
  factory Post({
    required String userName,
    required String userId,
    required String userPhotoUrl,
    required String text,
    required String name,
    required int commentCount,
    required List<String> likedUserIds,
    required List<String> tags,
    required DateTime createdDate,
    required List<DateTime> timeSlots,
    required List<String> images,
    required String id,
  }) = _Post;
}
